### PR TITLE
fix(auth-server): check for duplicate subs when creating subscription

### PR DESCRIPTION
**Pull request by julianpoy**
## Because

* Only the frontend enforces that you should not be able to create duplicate subscriptions.

## This pull request

* Checks to see if the subscription you're creating is a duplicate.

## Issue that this pull request solves

Closes FXA-6035

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts [hunk 1]
```diff
@@ -503,6 +511,25 @@ export class StripeHandler {
         metricsContext,
       } = request.payload as Record<string, string>;
 
+      // Make sure to clean up any subscriptions that may be hanging with no payment
+      const existingSubscription =
+        this.stripeHelper.findCustomerSubscriptionByPlanId(customer, priceId);
+      if (
+        existingSubscription &&
+        existingSubscription.status === 'incomplete'
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts:519]: Why not `existingSubscription?.status === 'incomplete'`?

File: packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts [hunk 2]
```diff
@@ -503,6 +511,25 @@ export class StripeHandler {
...
+      ) {
+        this.stripeHelper.cancelSubscription(existingSubscription.id);
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts:521]: Do we want to wait on this to make sure its actually removed before proceeding?

Comment (julianpoy) [reply packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts:521]: Was following a similar pattern to https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/payments/stripe.ts#L582.

Regardless, have updated this to wait for the promise to resolve.

File: packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts (hunk 1 above)
Comment (julianpoy) [reply packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts]: Sure. Updated.

File: packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts (hunk 2 above)
Comment (bbangert) [reply packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts]: Ah, that other one should prolly be awaited as well.

**Review by bbangert — APPROVED** [review]

**Pull request merged**